delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/model/StorageState.scala

```diff
@@ -81,17 +85,60 @@ object StorageState {
       updatedBy: Subject
   ) extends StorageState {
 
-    def storage: Storage =
+    def storage(cfg: StorageTypeConfig): Storage =
       value match {
-        case DiskStorageValue(default, algorithm, volume, read, write, maxSize)                       =>
-          DiskStorage(id, project, default, algorithm, volume, read, write, maxSize, tags, source)
-        case S3StorageValue(default, algorithm, bucket, settings, read, write, maxSize)               =>
-          S3Storage(id, project, default, algorithm, bucket, settings, read, write, maxSize, tags, source)
-        case RemoteDiskStorageValue(default, algorithm, endpoint, cred, folder, read, write, maxSize) =>
-          RemoteDiskStorage(id, project, default, algorithm, endpoint, cred, folder, read, write, maxSize, tags, source)
+        case DiskStorageValue(default, volume, read, write, maxSize)                                       =>
+          DiskStorage(
+            id = id,
+            project = project,
+            default = default,
+            algorithm = cfg.disk.digestAlgorithm,
+            volume = volume,
+            readPermission = read.getOrElse(cfg.disk.readPermission),
+            writePermission = write.getOrElse(cfg.disk.writePermission),
+            maxFileSize = maxSize.getOrElse(cfg.disk.maxFileSize),
+            tags = tags,
+            source = source
+          )
+        case S3StorageValue(default, bucket, endpoint, accessKey, secretKey, region, read, write, maxSize) =>
+          S3Storage(
+            id = id,
+            project = project,
+            default = default,
+            algorithm = cfg.amazonUnsafe.digestAlgorithm,
+            bucket = bucket,
+            endpoint = endpoint.orElse(cfg.amazonUnsafe.defaultEndpoint),
+            accessKey = accessKey.orElse(if (endpoint.forall(endpoint.contains)) cfg.amazonUnsafe.accessKey else None),
```

Comment:
If someone removes the aws config because it is not relevant at some point, we should be able to fetch the state even if it is deprecated

Reply:
I don't know what you mean

Reply:
1) Someone configure delta with s3, create storages relying on default values
2) S3 is not used anymore, the s3 config is removed
3) the amazonUnsafe will fail

Reply:
I understand your concern. I addressed that by having yet another  case class: `StorageFields`.
1. StorageFields: used as client input, to serialize json-ld into a case class and in `StorageCommand`
2. StorageValue: used for `StorageEvent`. It contains all the information from the `StorageFields` plus the resolved configuration
3. Storage: It contains `StorageValue` plus some extra fields like source, tags, id and project